core/drivers/crypto/hisilicon/sec_pbkdf2.c

```diff
@@ -0,0 +1,348 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright (c) 2024, HiSilicon Technologies Co., Ltd.
+ * Kunpeng hardware accelerator pbkdf2 implementation.
+ */
+
+#include <tee/tee_cryp_pbkdf2.h>
+#include <tee/tee_cryp_utl.h>
+
+#include "sec_main.h"
+#include "sec_pbkdf2.h"
+
+static enum hisi_drv_status sec_pbkdf2_parse_sqe(void *bd, void *msg __unused)
+{
+	struct hisi_sec_sqe *sqe = bd;
+	uint16_t done;
+
+	done = SEC_GET_FIELD(sqe->type2.done_flag, SEC_DONE_MASK, 0);
+	if (done != SEC_HW_TASK_DONE || sqe->type2.error_type) {
+		EMSG("SEC do pbkdf2 fail! done=0x%x, etype=0x%x",
+		     done, sqe->type2.error_type);
+		return HISI_QM_DRVCRYPT_IN_EPARA;
+	}
+
+	return HISI_QM_DRVCRYPT_NO_ERR;
+}
+
+static enum hisi_drv_status sec_pbkdf2_fill_sqe(void *bd, void *msg)
+{
+	struct sec_pbkdf2_msg *pbkdf2_msg = msg;
+	struct hisi_sec_sqe *sqe = bd;
+
+	sqe->type_auth_cipher = BD_TYPE2;
+	sqe->type_auth_cipher |= SHIFT_U32(AUTH_MAC_CALCULATE, SEC_AUTH_OFFSET);
+	sqe->sds_sa_type = SHIFT_U32(SCENE_PBKDF2, SEC_SCENE_OFFSET);
+
+	sqe->type2.mac_key_alg = SHIFT_U32(pbkdf2_msg->derive_type,
+					   SEC_AEAD_ALG_OFFSET);
+	/* mac_len = 1 and a_key_len = 1 only for hardware check */
+	sqe->type2.mac_key_alg |= 0x1;
+	sqe->type2.mac_key_alg |= SHIFT_U32(0x1, SEC_AKEY_OFFSET);
+
+	sqe->type2.alen_ivllen = pbkdf2_msg->salt_len;
+	sqe->type2.clen_ivhlen = pbkdf2_msg->c_num;
+	sqe->type2.pass_word_len = (uint16_t)pbkdf2_msg->key_len;
+	sqe->type2.dk_len = (uint16_t)pbkdf2_msg->out_len;
+
+#ifdef CFG_CRYPTO_HW_PBKDF2_WITH_EFUSE
```

Comment:
Prefer use `if (IS_ENABLED(CFG_CRYPTO_HW_PBKDF2_WITH_EFUSE))`
Ditto at line 94.

Reply:
All review comments have been modified.

Reply:
Hi @etienne-lms, If it's okay, can you give me an Acked-by?